sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -54,6 +54,10 @@ class SubquerySuite extends QueryTest with SharedSQLContext {
     t.createOrReplaceTempView("t")
   }
 
+  test("rdd deserialization does not crash [SPARK-15791]") {
+    sql("select (select 1 as b) as b").rdd.count()
```

Comment:
+1 

although not that big of a deal, since other tests already cover some correctness. I'm going to merge this one to cut down the number of 2.0 blockers.